cfme/tests/infrastructure/test_host.py

```diff
@@ -248,3 +248,63 @@ def test_250_vmware_hosts_loading(appliance, create_250_hosts, view_type):
     view.entities.paginator.set_items_per_page(1000)
     view.toolbar.view_selector.select(view_type)
     wait_for(view.entities.get_first_entity, timeout=60, message='Wait for the view')
+
+
+@test_requirements.general_ui
+@pytest.mark.provider([RHEVMProvider], selector=ONE, scope="module", override=True)
+@pytest.mark.parametrize(
+    "power_state", ["preparing_for_maintenance", "maintenance", "unknown", "off", "on"]
+)
+def test_infrastructure_hosts_icons_states(
+    appliance, request, power_state, setup_provider, provider, soft_assert
+):
+    """
+    Polarion:
+        assignee: pvala
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/3h
+        setup:
+            1. Add a RHEVM provider.
+            2. SSH into appliance console and run `psql vmdb_production`
+        testSteps:
+            1. Check if the Quadicon(Host's ALL page)
+                and host(Host's Detail page) power_state changes after running the command:
+                `UPDATE hosts SET power_state = ':power_state' WHERE name=':host_name';`
+    """
+    # get host and host details
+    host = provider.hosts.all()[0]
+    host_name = host.name
+    reset_state = host.rest_api_entity.power_state
+
+    command = (
+        "psql vmdb_production -c \"UPDATE hosts SET power_state='{state}'"
```

Comment:
This must be done through `appliance.db` object.

Test pattern is good though - we don't actually want to try and modify the host states, so directly injecting the states that we want to test into the DB is a good alternative.